### DISCUSSION TITLE
how to access DLT Daemon buffers / cache

### DISCUSSION THREAD

**Discussion by GuruDlt** [COVESA/dlt-daemon#710] [General]
Hello All,

We tried to fetch DLT logs through custom dlt client (loop back interface based socket connection) with DLT daemon, but it provides one log at a time with call back which is causing performance issues with load on cpu.
So I m looking for a way to fetch "last X logs" from DLT daemon using its buffer if it provides?
Does DLT Daemon provides any logged buffers? 
if it provides then how to access DLT daemon logged buffers or cache through dlt client or any other way which is possible.

**Comment by minminlittleshrimp** [comment]
By connect a client via ipc, it is automatically fetch logs from DltClient buffer from daemon.
However, this is flaw design of single thread logging, so callback is in sequence and might be blocked by a long fetching client with low speed during buffer filled, making it always blocked, or even closed socket (in src code noted: Fixme: why the he** we close socket here)
Team is planning for the design mitigate and for now not yet proceed due to protocol v2 PR under working
